packages/radix-ui-themes/src/components/text-field.css

```diff
@@ -1,45 +1,104 @@
 .rt-TextFieldRoot {
   display: flex;
-  box-sizing: border-box;
-  position: relative;
-  z-index: 0;
+  align-items: stretch;
+
+  /* Make sure that font weight is not inherited, e.g. from a wrapping <label> */
+  font-weight: var(--font-weight-regular);
+
+  @supports selector(:has(*)) {
+    &:where(:has(.rt-TextFieldInput:focus)) {
+      outline: 2px solid var(--text-field-focus-color);
+      outline-offset: -1px;
+    }
+  }
+  @supports not selector(:has(*)) {
+    &:where(:focus-within) {
```

Comment:
Why not only do it via focus-within? It has better support than `:has` anyway. Or is that because if has wasn't supported and we'd use this, it would get focused when buttons in slots are focused?